src/nci_util.c

```diff
@@ -402,6 +421,66 @@ nci_parse_iso_dep_poll_a_param(
     return FALSE;
 }
 
+static
+gboolean
+nci_parse_iso_dep_poll_b_param(
+    NciActivationParamIsoDepPollB* param,
+    const guint8* bytes,
+    guint len)
+{
+    /* NFCFrum-TS-NCI-1.0
+     * Table 75: Activation parameters for NFC-B/ISO-DEP Poll Mode
+     *
+     * +============================================================+
+     * | Offset | Size | Description                                |
+     * +============================================================+
+     * | 0      | 1    | Length of ATTRIB Response Parameter (n)    |
+     * | 1      | n    | ATTRIB Response                            |
+     * +============================================================+
+     */
+
+    /* ATTRIB Response */
+    const guint attrib_length = bytes[0];
+
+    if (attrib_length >= 1) {
+        /* NFCForum-TS-DigitalProtocol-1.01
+         * Table 79: ATTRIB Response Format */
+
+#define NFC_T4B_MBLI_MASK (0xF0) /* MBLI Mask */
+#define NFC_T4B_DID_MASK  (0x0F) /* DID Mask */
+
+        /* First two octets of the first byte of ATTRIB Response */
+        param->mbli = (bytes[1] & NFC_T4B_MBLI_MASK) >> 4;
+        param->did  = (bytes[1] & NFC_T4B_DID_MASK);
+
+        /* Higher Layer Response */
+        if (attrib_length >= 2) {
+            param->hilr.bytes = bytes + 2;
+            param->hilr.size = attrib_length - 1;
```

Comment:
This block is not covered by unit tests, I think it makes sense to add a case for it too. There are examples in `test_nci_util.c` how that could be done, when `GUtilData` points back to a part of the input buffer.

Reply:
Done. Now tests have two cases with zero length HILR (with data from an actual tag) and with a non-zero length HILR (some bollocks data as it seems to be application specific and depends on command sent by NFC chip).